packages/contracts/test/LibExperience.t.sol

```diff
@@ -27,4 +28,64 @@ contract LibExperienceTest is MudV2Test {
       assertEq(exp[i], 1);
     }
   }
-}
\ No newline at end of file
+
+  function testGetExp() public {
+    LibExperience.initExp(targetEntity);
+    assertTrue(LibExperience.hasExp(targetEntity));
+
+    uint32[PStat_length] memory addExp = [uint32(1), 1, 1];
+    LibExperience.increaseExp(targetEntity, addExp);
+
+    uint32[PStat_length] memory exp = LibExperience.getExp(targetEntity);
+    for (uint256 i; i < exp.length; i++) {
+      assertEq(exp[i], 1);
+    }
+  }
+
+  function testHasExp() public {
+    assertFalse(LibExperience.hasExp(targetEntity));
+
+    LibExperience.initExp(targetEntity);
+    assertTrue(LibExperience.hasExp(targetEntity));
+  }
+
+  function testSetExp() public {
+    uint32[PStat_length] memory exp = [uint32(10), 20, 30];
+    Experience.set(targetEntity, exp);
+
+    uint32[PStat_length] memory retrievedExp = LibExperience.getExp(targetEntity);
+    for (uint256 i; i < retrievedExp.length; i++) {
+      assertEq(retrievedExp[i], exp[i]);
+    }
+  }
+
+  function testAddExp() public {
+    LibExperience.initExp(targetEntity);
+    assertTrue(LibExperience.hasExp(targetEntity));
+
+    uint32[PStat_length] memory initialExp = LibExperience.getExp(targetEntity);
+
+    uint32[PStat_length] memory addExp = [uint32(5), 10, 15];
+    LibExperience.increaseExp(targetEntity, addExp);
+
+    uint32[PStat_length] memory expectedExp;
+    for (uint256 i; i < initialExp.length; i++) {
+      expectedExp[i] = initialExp[i] + addExp[i];
+    }
+
+    uint32[PStat_length] memory retrievedExp = LibExperience.getExp(targetEntity);
+    for (uint256 i; i < retrievedExp.length; i++) {
+      assertEq(retrievedExp[i], expectedExp[i]);
+    }
+  }
+
+  function testMaxLevel() public {
+    LibExperience.initExp(targetEntity);
+
+    // Set the experience to the maximum level
+    uint32[PStat_length] memory maxExp = [type(uint32).max, type(uint32).max, type(uint32).max];
+    Experience.set(targetEntity, maxExp);
+
+    assertEq(LibExperience.getAggregateLevel(targetEntity, maxExp), 16);
+  }
```

Comment:
this just tests that an exp cap exists,
you could also test for specific pstat values using `getExpForPStats`,
fuzz tests for pstat values might also be nice